### ISSUE TITLE
Completion hangs in Emacs/lsp-latex

### ISSUE THREAD

**Issue by d9d6ka** [latex-lsp/texlab#802]
The problem is the same as in #456.

I use texlab in `Emacs` (Windows build) with `lsp-mode`.
Emacs version: 28.2
LSP-mode: latest from git (installed via straight)
texlab version: 4.3.0 (installed via Scoop).

Freezes start immediately after the document is opened. Emacs uses 100% its CPU core (Windows builds are single threaded). It takes Emacs 20-30 seconds on my i5 machine to respond after the autocompletion call. Fter the initial freeze I have from 0 to 2 minutes of normal work.
Interestingly, there is no freezes in Emacs 28.1 (checked on the official Windows build from GNU ftp).

**Comment by pfoerster** [comment]
@d9d6ka Can you provide a log file  (using `texlab -vvvv --log-file ...`), please? The previous version of this issue was caused by a configuration request that was not answered by Emacs.

**Comment by d9d6ka** [comment]
How should I get it? I've tried [this](https://github.com/ROCKTAKEY/lsp-latex/issues/30#issuecomment-875658487), but it doesn't work (maybe I should have lsp-latex installed).

**Comment by pfoerster** [comment]
Thanks for the log file. There are some things going on, which do not seem right:

- Emacs is sending a lot of `textDocument/documentSymbols` and `textDocument/documentLink` requests. 
In fact, there are so many that `texlab` cannot keep up with responding to them in-time. There is one section in the log file, where Emacs sends 7 `textDocument/documentLink` requests although the document has not changed in the mean time. Document symbols are particulary expensive because the whole syntax tree of the file needs to be traversed and the output can be quite large compared to other requests.
- Emacs is cancelling finished requests. However, this is not a problem for `texlab` as the server does not support cancellation so the messages are discarded.
- The colon after the drive letter is URL-encoded in incoming requests from Emacs (likely not the problem and easy to handle in `texlab`)

So the TLDR version is that Emacs sends way too many unnecessary requests which slow down `texlab`.

**Comment by pfoerster** [comment]
@d9d6ka Are there updates on this issue? Does this problem still occur?

**Comment by d9d6ka** [comment]
> @d9d6ka Are there updates on this issue? Does this problem still occur?

Yes, it seems so. I haven't bumped into the issue for a long time already. Everything works as expected! Thanks!
